src/main/java/uk/gov/companieshouse/filinghistory/api/service/FilingHistoryPutRequestValidator.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.filinghistory.api.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.ExternalData;
+import uk.gov.companieshouse.api.filinghistory.InternalData;
+import uk.gov.companieshouse.api.filinghistory.InternalFilingHistoryApi;
+
+@Component
+public class FilingHistoryPutRequestValidator<T> implements Validator<T> {
```

Comment:
Can use the concrete type as type parameter when implementing the interface instead of casting in the method. 
```public class FilingHistoryPutRequestValidator implements Validator<InternalFilingHistoryApi> {```